### DISCUSSION TITLE
Two-Way Server-Side Bucket Replication with NewerNoncurrentVersions lifecycle rule

### DISCUSSION THREAD

**Discussion by szabokacsa** [minio/minio#14069] [Q&A] [answered]
Hi,
I have a Two side (bi-direstional ) replication minio setup.
I wanted to delete the older versions on both cluster with a new NewerNoncurrentVersions  lifecycle Rule in order to free up space.
I created this Rule only on one cluster, and I expected the cleanup will happen on both cluster because of the replication.
My problem is that the clean up only happens on the cluster where I created a lifecycle, but it has no effect on the other cluster and I don't know why.

I created the replication base on this guide, with a modification that I'm usingonly one replication user where I merged the replication policy for the two replication user what the guide detailed : https://docs.min.io/minio/baremetal/replication/enable-server-side-two-way-bucket-replication.html

I attached the replication policy what I'm using.
[replication_config.txt](https://github.com/minio/minio/files/7837682/replication_config.txt)

This is content of the lifecycle Rule:
{
    "Rules": [
        {
            "ID": "Keep only most recent 1 noncurrent version",
            "Status": "Enabled",
            "NoncurrentVersionExpiration": {
                "NewerNoncurrentVersions": 1
            }
        }
    ]
}

**Comment by harshavardhana** [comment, marked as answer]
You need to set ILM rules on both source and target separately.

**Reply by ravindk89** [reply]
@szabokacsa as documented [here](https://docs.min.io/minio/baremetal/replication/replication-overview.html#replication-of-delete-operations), replication deletes do not trigger lifecycle management rules.

I'll make a note to ensure that information is sync'd to the active-active procedures as well.

**Reply by szabokacsa** [reply]
Okay understood, thanks you all for the response!
